src/repo/migrate/index.js

```diff
@@ -557,6 +557,71 @@ const migrate3xFrom13xto14x = async (db) => {
 };
 
 
+const migrate3xFrom14xto15x = async (db) => {
+    // remove the MIN constraint
+    await db.command(
+        'ALTER PROPERTY CdsPosition.pos MIN NULL',
+    ).all();
+
+    // update mis-parsed variants
+    const misParsed = await db.query(
+        `SELECT
+            *
+        FROM
+            PositionalVariant
+        WHERE
+            (
+                (
+                    break1Start.@class == 'CdsPosition'
+                    AND break1Start.pos == 1
+                    AND break1Start.offset < 0
+                )
+                OR (
+                    break2Start.@class == 'CdsPosition'
+                    AND break2Start.pos == 1
+                    AND break2Start.offset < 0
+                )
+                OR (
+                    break2End.@class == 'CdsPosition'
+                    AND break2End.pos == 1
+                    AND break2End.offset < 0
+                )
+                OR (
+                    break1End.@class == 'CdsPosition'
+                    AND break1End.pos == 1
+                    AND break1End.offset < 0
+                )
+            )
+            AND deletedAt IS NULL`,
+    ).all();
+    logger.info(`found ${misParsed.length} variants to be updated`);
+
+    for (const variant of misParsed) {
+        const updated = { };
+
+        for (const breakPoint of ['break1Start', 'break2Start', 'break1End', 'break2End']) {
+            if (
+                variant[breakPoint]
+                && variant[breakPoint]['@class'] === 'CdsPosition'
+                && variant[breakPoint].pos === 1
+                && variant[breakPoint].offset < 0
+            ) {
+                updated[breakPoint] = { ...variant[breakPoint], offset: 0, pos: variant[breakPoint].offset };
+            }
+        }
+
+        for (const repr of ['break1Repr', 'break2Repr', 'displayName']) {
+            if (variant[repr]) {
+                updated[repr] = variant[repr].replace(/([._])1(-\d+)/g, '$1$2');
```

Comment:
what change does this make?

Reply:
this is for the cds variants that would have been misparsed by the old parser. Previously the parser would have transformed `c.-42G>T` into `c.1-42G>T`. So this corrects that and sets the pos as the offset and fixes the strings

Reply:
specifically it's the strings I'm wondering about, it looks like it takes the 1 out ?

Reply:
yes that's correct. So it puts the `c.1-42G>T` back to what it should be as `c.-42G>T`

Reply:
You can test this migration by starting the dev server locally and pointing it to either dev or staging db